src/OrbitGl/PrimitiveAssembler.cpp

```diff
@@ -302,6 +302,14 @@ void PrimitiveAssembler::AddQuadBorder(const Quad& quad, float z, const Color& c
   AddLine(quad.vertices[3], quad.vertices[0], z, color);
 }
 
+void PrimitiveAssembler::AddAabbOutline(Vec2 pos, Vec2 size, float outline_width, float z,
```

Comment:
Could these both be `const Vec&` parameters?

Reply:
The `Vec2` type is so simple, that we probably rather want the copy here.

Reply:
Vec2 should be passed in register, it's better to keep as-is.

Reply:
FWIW, Vec2 has a custom destructor, therefore it's considered non-trivial, and AFAIK that means it can't be passed in a registers.

From the Itanium ABI spec:

> If a parameter type is a class type that is [non-trivial for the purposes of calls](https://itanium-cxx-abi.github.io/cxx-abi/abi.html#non-trivial), the caller must allocate space for a temporary and pass that temporary by reference.

and

> A type is considered non-trivial for the purposes of calls if: it has a non-trivial copy constructor, move constructor, or destructor, or all of its copy and move constructors are deleted.



To fix that we should replace GteVector at some point.